addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -143,12 +154,219 @@ odoo.define('website_form.s_website_form', function (require) {
             // Reinitialize dates
             this.$allDates.removeClass('s_website_form_datepicker_initialized');
         },
+        /**
+        * calculates the visibility at each event input on the form
+        * @param {Object} event
+        */
+        onFormInputChange: function (event) {
+            document.querySelectorAll('[data-visibility-dependency]').forEach((input)=>{
+                const idOfDependency = input.getAttribute('data-visibility-dependency');
+                if (idOfDependency) {
+                    const dependecyField = document.getElementById(idOfDependency);
+                    if (dependecyField) {
+                        this._manageVisibility(dependecyField, input);
+                    }
+                }
+            });
+        },
+        /**
+        * checks the data visibility condition and show/hide field
+        *
+        * @param {HTMLElement} dependency the field that is concerned by the visibility condition
+        * @param {HTMLElement} input the input to hide or to show depends of visibility condition
+        */
+        _manageVisibility: function (dependency, input) {
+            if (!input) {
+                return;
+            }
+            let field = dependency;
+            while (!field.classList.contains('s_website_form_field')) {
+                field = field.parentElement;
+            }
+            if (field.classList.contains('s_website_form_field_hidden_if') && field.classList.contains('hidden')) {
+                //The dependecy is hidden so this field is hidden too
+                return this._hideField(input);
+            }
+            const visibilityCondition = input.getAttribute('data-visibility-condition');
+            const comparator = input.getAttribute('data-visibility-comparaison');
+            if (['radio', 'checkbox'].includes(dependency.type)) {
+                dependency = document.getElementById(visibilityCondition);
+            }
+            const currentValue = this._readValueOfInput(dependency);
+            if (dependency.type === 'checkbox' || dependency.type === 'radio') {
```

Comment:
I haven't found an easy way to find the fields with their id via the form element API.

Reply:
I made some tests with `FormData` API and think there should be something to use on that side: https://jsfiddle.net/onuqwh9s

- I learned that a disabled element is not sent to the server via submit and is not considered for FormData etc. I think your code should handle that: the invisible elements are not just invisible: they should be disabled, this is actually exactly what you need to discard the elements to not send (don't know if this is what you did). That way if a field depends on the field "hello", you can always retrieve its value by name, not by id, whatever the field type and using the JavaScript API instead of your custom code.

- For the previous point to work, inputs with the same name should be synchronized I guess but it actually make sense as a feature: if the form contains two inputs for the same field "What is your age?", if I fill in the form with "29" then changes something in the form that makes that input disappear but later makes the *other* age input appear... I expect it to still be filled with "29" even though it is not the same input element (it is the same field).

Reply:
In the continuity of what you say I think I should add, the fact that a field is visible only if the other fields that have the same name are invisible. This native visibility condition would be an additional condition to the one composed by the user.